Dockerfile-ngc

```diff
@@ -0,0 +1,201 @@
+		unattended-upgrades \
+	&& unattended-upgrade \
+	&& rm -rf /var/lib/apt/lists/* \
+	&& rm /etc/ssh/ssh_host_ecdsa_key \
+	&& rm /etc/ssh/ssh_host_ed25519_key \
+	&& rm /etc/ssh/ssh_host_rsa_key \
+   && ln -s /usr/include/slurm-wlm /usr/include/slurm
+		
+COPY dockerfile_scripts /tmp/det_dockerfile_scripts
+
+# Try overriding the default GCC to an older one to hopefully enable
+# horovod to compile.
+#RUN apt-get update \
+#    && DEBIAN_FRONTEND=noninteractive apt-get install -y gcc-9 g++-9 \
+#    && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 100 \
+#    --slave /usr/bin/g++ g++ /usr/bin/g++-9 --slave /usr/bin/gcov gcov \
+#    /usr/bin/gcov-9
+
+
+# Install debuild util, etc. for later compiling GDRcopy libraries
+# RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y devscripts debhelper
+
+# Note for the NGC GH image the libgdr is already installed so don't need this
+# Install GDRcopy so OMPI, NCCL, AWS plugin, etc, can see it
+# RUN /tmp/det_dockerfile_scripts/build_gdrcopy.sh
+
+ARG WITH_MPI
+ARG WITH_OFI
+ARG UCX_INSTALL_DIR=/container/ucx
+ARG OMPI_INSTALL_DIR=/container/ompi
+ARG OFI_INSTALL_DIR=/container/ofi
+ARG OMPI_WITH_CUDA=1
+RUN if [ "$WITH_MPI" = "1" ]; then /tmp/det_dockerfile_scripts/ompi.sh "$UBUNTU_VERSION" "$WITH_OFI" "$OMPI_WITH_CUDA"; fi
+# Make sure OMPI/UCX show up in the right paths
+ARG VERBS_LIB_DIR=/usr/lib/libibverbs
+ARG UCX_LIB_DIR=${UCX_INSTALL_DIR}/lib:${UCX_INSTALL_DIR}/lib64
+ARG UCX_PATH_DIR=${UCX_INSTALL_DIR}/bin
+ARG OFI_LIB_DIR=${OFI_INSTALL_DIR}/lib:${OFI_INSTALL_DIR}/lib64
+ARG OFI_PATH_DIR=${OFI_INSTALL_DIR}/bin
+ARG OMPI_LIB_DIR=${OMPI_INSTALL_DIR}/lib
+ARG OMPI_PATH_DIR=${OMPI_INSTALL_DIR}/bin
+
+# Set up UCX_LIBS and OFI_LIBS
+ENV UCX_LIBS="${VERBS_LIB_DIR}:${UCX_LIB_DIR}:${OMPI_LIB_DIR}:"
+ENV OFI_LIBS="${VERBS_LIB_DIR}:${OFI_LIB_DIR}:${OMPI_LIB_DIR}:"
+
+# If WITH_OFI is true, then set EXTRA_LIBS to OFI libs, else set to empty string
+ENV EXTRA_LIBS="${WITH_OFI:+${OFI_LIBS}}"
+
+# If EXTRA_LIBS is empty, set to UCX libs, else leave as OFI libs
+ENV EXTRA_LIBS="${EXTRA_LIBS:-${UCX_LIBS}}"
+
+# But, only add them if WITH_MPI
+ENV LD_LIBRARY_PATH=${WITH_MPI:+$EXTRA_LIBS}:$LD_LIBRARY_PATH
+
+#USING OFI
+#ENV PATH=${WITH_OFI:+$PATH:${WITH_MPI:+$OFI_PATH_DIR:$OMPI_PATH_DIR}}
+ENV PATH=${WITH_OFI:+${WITH_MPI:+$OFI_PATH_DIR:$OMPI_PATH_DIR}:$PATH}
+ENV PATH=/container/ompi/bin:$PATH
+ENV LD_LIBRARY_PATH=/container/ompi/lib:$LD_LIBRARY_PATH
+#RUN which mpicxx
+
+
+#USING UCX
+#ENV PATH=${PATH:-$CONDA:${WITH_MPI:+$UCX_PATH_DIR:$OMPI_PATH_DIR}}
+
+# Enable running OMPI as root
+ENV OMPI_ALLOW_RUN_AS_ROOT ${WITH_MPI:+1}
+ENV OMPI_ALLOW_RUN_AS_ROOT_CONFIRM ${WITH_MPI:+1}
+
+# Install TensorFlow
+ARG TENSORFLOW_PIP
+#RUN if [ "$TENSORFLOW_PIP" ]; then \
+#		export HOROVOD_WITH_TENSORFLOW=1 \
+#		&& python -m pip install $TENSORFLOW_PIP; \
+#	else \
+#		export HOROVOD_WITH_TENSORFLOW=0; \
+#	fi
+#RUN python -m pip install tensorflow-gpu
+
+# Install Horovod

```

Comment:
Since Horovod is not being installed here because it does not work with the newest PyTorch, might be cleaner to remove these lines. The py3.8 version of the build targets will have the Horovod.